### ISSUE TITLE
Add data for Cross Cluster Search compatibility problems for upgrade API

### ISSUE THREAD

**Issue by LeeDr** [elastic/elasticsearch#83429]
### Description

If a "local" 7.17 cluster has remote clusters defined for Cross Cluster Search (CCS) which are older than version 7.17, and they upgrade to 8.0, this local cluster will no longer connect to those remote clusters.

Elasticsearch will log a message like this;
```
Received message from unsupported version: [7.15.2] minimal compatible version is: [7.17.0]
```

If possible, we should update the upgrade API to let the Kibana Upgrade Assistant (UA) know so it can show appropriate warnings before a user upgrades.
And Cloud can also use the response to that API to block users from upgrading in this case.

I *think* we should guide users to upgrade the remote clusters first?  I haven't tested 7.17 with remote 8.0 clusters yet but since Kibana 7.17 has been tested with Elasticsearch 8.0 I think this is a supported direction.

**Comment by dakrone** [comment]
> we should update the upgrade API

We've noticed a bit of a disconnect between these APIs, the API is actually the deprecation API, rather than the upgrade-ready API. There is a bit of tension between its intended use for displaying deprecations and using it as a "upgrade-ready" API. For example, another concern is in https://github.com/elastic/elasticsearch/issues/82807 where the disk space is checked.

I wonder if it makes sense to do this check from the upgrade assistant rather than inside the Elasticsearch API itself, as the UI is explicit about it being used as part of the upgrade process rather than explicitly about deprecated things.

**Comment by cjcenizal** [comment]
> do this check from the upgrade assistant 

Worth exploring. I have a more fundamental question, though. Do we have the ability to retrieve remote cluster Stack version? I looked at the Get Remote info API docs and don't see that info in the response. Can someone from @elastic/es-distributed tell us if this is currently available in an ES API and, if not, if it's possible to add this to the API?

**Comment by cjcenizal** [comment]
I just realized that if this isn't already available, then our options for _making_ it available in the API are limited because we'd probably have to ship the change in another version... which would require upgrading. 🤦

**Comment by dakrone** [comment]
I looked into this a little bit, this information is not straightforward to retrieve at all, because the versions aren't something we currently store and can access for remote clusters.

On thinking about it a bit more, I think it's going to be difficult for the UI to get this information also, because they would have to try to connect to the server, which only specifies transport ports anyway.

That said, I think this would be a pretty invasive change to try and support something like this for a 7.17.x point release, so I don't think I'm in favor of trying to tackle this currently. Perhaps we can warn the user in a different way, such as the UI checking for the presence of remote clusters configured, and if present, warn the user that they will also need to upgrade those clusters to be able to use CCS/CCR functionality. How do people feel about that as a solution?

**Comment by cjcenizal** [comment]
@LeeDr That seems like a good solution to me. @LeeDr WDYT?

CC @elastic/platform-deployment-management for your thoughts too

**Comment by LeeDr** [comment]
Yes, I think if UA could just display a message and a link to docs if they have remote clusters defined that would be pretty good.  I think we also need some docs that give clear info on it and I don't think we have that ATM.

**Comment by cjcenizal** [comment]
Transferring this to the Kibana repo since this work is now scoped to the UI.
